block/delete.go

```diff
@@ -51,6 +51,13 @@ outer_loop:
 		if header.Number < finalBlockNumber {
 			blockheader.Set(header.Number, digest, header.Version, header.Timestamp)
 			log.Infof("finish: _NOT_ Deleting: %d", header.Number)
+
+			nextDifficulty, prevDifficulty, err := blockrecord.AdjustDifficultyAtBlock(header.Number)
+			if err != nil {
+				log.Criticalf("failed to adjust difficulty with error: %s", err)
```

Comment:
if really critical the require a logger.Panicf(…) showing the same output - this will terminate the program.
If not really a fatal error, then Change to Errorf(…)

Reply:
changed